src/darwin/Framework/CHIP/MTRConversion.h

```diff
@@ -38,7 +38,8 @@ AsNumber(chip::Optional<T> optional)
 
 inline NSDate * MatterEpochSecondsAsDate(uint32_t matterEpochSeconds)
 {
-    return [NSDate dateWithTimeIntervalSince1970:(chip::kChipEpochSecondsSinceUnixEpoch + (NSTimeInterval) matterEpochSeconds)];
+    const uint64_t interval = static_cast<uint32_t>(chip::kChipEpochSecondsSinceUnixEpoch) + matterEpochSeconds;
```

Comment:
This change also looks wrong: adding two uint32_t will give you another uint32_t, possibly with overflow.  This should probably cast kChipEpochSecondsSinceUnixEpoch to uint64_t?  Or just cast both to NSTimeInterval.